src/bpftrace.cpp

```diff
@@ -39,6 +39,37 @@ namespace bpftrace {
 DebugLevel bt_debug = DebugLevel::kNone;
 bool bt_verbose = false;
 
+int format(char * s, size_t n, const char * fmt, std::vector<std::unique_ptr<IPrintable>> &args) {
+  switch(args.size()) {
+    case 0:
+      return snprintf(s, n, fmt);
+      break;
+    case 1:
+      return snprintf(s, n, fmt, args.at(0)->value());
+      break;
+    case 2:
+      return snprintf(s, n, fmt, args.at(0)->value(), args.at(1)->value());
+      break;
+    case 3:
+      return snprintf(s, n, fmt, args.at(0)->value(), args.at(1)->value(), args.at(2)->value());
+      break;
+    case 4:
+      return snprintf(s, n, fmt, args.at(0)->value(), args.at(1)->value(), args.at(2)->value(), args.at(3)->value());
+      break;
+    case 5:
+      return snprintf(s, n, fmt, args.at(0)->value(), args.at(1)->value(), args.at(2)->value(),
+        args.at(3)->value(), args.at(4)->value());
+      break;
+    case 6:
+      return snprintf(s, n, fmt, args.at(0)->value(), args.at(1)->value(), args.at(2)->value(),
+        args.at(3)->value(), args.at(4)->value(), args.at(5)->value());
+      break;
+    default:
+      std::cerr << "printf() can only take up to 7 arguments (" << args.size() << ") provided" << std::endl;
```

Comment:
```suggestion
      std::cerr << "format() can only take up to 7 arguments (" << args.size() << ") provided" << std::endl;
```
Old code was wrong too, but let's fix it